src/Server/RestRouterHandlers/TableStatsRestRouterHandler.cpp

```diff
@@ -0,0 +1,253 @@
+#include "TableStatsRestRouterHandler.h"
+
+#include <DistributedMetadata/CatalogService.h>
+#include <DistributedMetadata/sendRequest.h>
+#include <Storages/DistributedMergeTree/StorageDistributedMergeTree.h>
+
+namespace DB
+{
+namespace ErrorCodes
+{
+    extern const int OK;
+    extern const int BAD_REQUEST_PARAMETER;
+    extern const int UNKNOWN_TABLE;
+    extern const int TYPE_MISMATCH;
+    extern const int UNKNOWN_EXCEPTION;
+}
+
+namespace
+{
+const String TABLE_STATS_URL = "http://{}:{}/dae/v1/tablestats/{}?local={}";
+
+String buildResponse(const String & query_id, Poco::JSON::Object & resp)
+{
+    resp.set("request_id", query_id);
+    std::stringstream resp_str_stream; /// STYLE_CHECK_ALLOW_STD_STRING_STREAM
+    resp.stringify(resp_str_stream, 0);
+
+    return resp_str_stream.str();
+}
+}
+
+std::pair<String, Int32> TableStatsRestRouterHandler::executeGet(const Poco::JSON::Object::Ptr & /* payload */) const
+{
+    const auto & table = getPathParameter("table");
+    if (table.empty())
+        return {jsonErrorResponse("Table is empty.", ErrorCodes::BAD_REQUEST_PARAMETER), HTTPResponse::HTTP_BAD_REQUEST};
+
+    /// check table validity
+    const auto & catalog_service = CatalogService::instance(query_context);
+    const auto & table_ptrs = catalog_service.findTableByName(database, table);
+    if (table_ptrs.empty())
+        return {jsonErrorResponse(fmt::format("TABLE '{}.{}' does not exist.", database, table), ErrorCodes::UNKNOWN_TABLE),
+                HTTPResponse::HTTP_BAD_REQUEST};
+
+    assert(table_ptrs[0]);
+    if (table_ptrs[0]->engine != "DistributedMergeTree")
+        return {jsonErrorResponse(fmt::format("No support TABLE '{}.{}' Engine {}.", database, table, table_ptrs[0]->engine), ErrorCodes::TYPE_MISMATCH),
+                HTTPResponse::HTTP_BAD_REQUEST};
+
+    try
+    {
+        /// if local == true, only local table stats
+        if (getQueryParameterBool("local", false))
+        {
+            Poco::JSON::Object resp;
+            int err = buildLocalTableStatsJSON(resp, table);
+            if (err != ErrorCodes::OK)
+            {
+                return {buildResponse(query_context->getCurrentQueryId(), resp), HTTPResponse::HTTP_INTERNAL_SERVER_ERROR};
+            }
+            return {buildResponse(query_context->getCurrentQueryId(), resp), HTTPResponse::HTTP_OK};
+        }
+
+        /// Forward all nodes where the table is shard/replica to get local table stats.
+        std::multimap<int, CatalogService::TablePtr> shard_tables;
+        for (auto tptr : table_ptrs)
+            shard_tables.emplace(tptr->shard, tptr);
```

Comment:
what happens if there are more than 1 replica in each shard?

Reply:
It's a multimap, so can be support one key - multi values, which is also the reason to used here

Reply:
Actually my question is that whether it is possible duplicated queries will be sent to different replicas in the same shard.